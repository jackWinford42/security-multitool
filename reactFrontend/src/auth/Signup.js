import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Alert, Card, CardBody } from 'reactstrap';
import "./formStyles.css"

/** Sign up form for getting the username, password, first name,
 * last name, and email of a new user. On submit a callback function
 * from app is called to register the user with the values from the
 * form.
 */
export default function Signup({ signup }) {
  const history = useHistory();
  console.debug("Sign up form");

  const [formData, setFormData] = useState({
    username:"",
    password:"",
    email:"",
    profile_pic:""
  });
  const [errors, setErrors] = useState([]);

  async function handleSubmit(evt) {
    evt.preventDefault();
    //send the form data to app's signup function
    const res = await signup(formData)
    if (res.worked) history.push("/");
    else setErrors(res.errors);
  }

  // Update form data to reflect change in form fields
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
  }

  return (
      <div className="SignupForm col-md-6 col-lg-4 offset-md-3 offset-lg-4">
        <Card className="authFormCard">
          <CardBody>
            <form className="form-inline" onSubmit={handleSubmit}>
              <label>Username:</label>
              <input
                className="form-control form-control-md"
                name="username"
                placeholder="username"
                value={formData.username}
                onChange={handleChange}
              />
              <label>Password:</label>
              <input
                className="form-control form-control-md"
                name="password"
                placeholder="password"
                value={formData.password}
                onChange={handleChange}
              />
              <label>Email:</label>
              <input
                className="form-control form-control-md"
                name="email"
                placeholder="email"
                value={formData.email}
                onChange={handleChange}
              />
              <label>Profile Picture Url:</label>
              <input
                className="form-control form-control-md"
                name="profile_pic"
                placeholder="url"
                value={formData.profile_pic}
                onChange={handleChange}
              />
              <button type="submit" className="btn btn-lg btn-primary">
                sign up
              </button>
            </form>
            {errors.length > 0 && 
            errors.map(error => <Alert color="danger" key={error}>{error}</Alert>)}
          </CardBody>
        </Card>
      </div>
  );
}