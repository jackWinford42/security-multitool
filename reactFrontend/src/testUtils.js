import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };