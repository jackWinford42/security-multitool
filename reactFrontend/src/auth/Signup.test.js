import React from "react";
import { render } from "@testing-library/react";
import Signup from "./Signup";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Signup/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Signup/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});