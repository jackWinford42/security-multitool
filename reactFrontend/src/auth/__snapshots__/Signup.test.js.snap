// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm col-md-6 col-lg-4"
  >
    <div
      class="authFormCard card"
    >
      <div
        class="card-body"
      >
        <form
          class="form-inline"
        >
          <label>
            Username:
          </label>
          <input
            class="form-control form-control-md"
            name="username"
            placeholder="username"
            value=""
          />
          <label>
            Password:
          </label>
          <input
            class="form-control form-control-md"
            name="password"
            placeholder="password"
            type="password"
            value=""
          />
          <label>
            Email:
          </label>
          <input
            class="form-control form-control-md"
            name="email"
            placeholder="email"
            value=""
          />
          <label>
            Profile Picture Url:
          </label>
          <input
            class="form-control form-control-md"
            id="bottomInput"
            name="profile_pic"
            placeholder="url"
            value=""
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained authButton MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              sign up
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
